src/components/ModalInput/ModalInput.tsx

```diff
@@ -88,7 +90,7 @@ const ModalInput: React.FC<ModalInputProps> = ({
         </Flex>
         <Flex alignItems="flex-end" justifyContent="space-around">
           <StyledInput
-            pattern="^[0-9]*[.,]?[0-9]*$"
+            pattern={`^[0-9]*[.,]?[0-9]{0,${decimals}}$`}
```

Comment:
This pattern (and decimal prop) should be updated on the UI Kit and removed from here.

Reply:
I opened a pr for BalanceInput in UI Kit

https://github.com/pancakeswap/pancake-toolkit/pull/142

Reply:
Input component is used in many places for different reasons in frontend (such as search input for text, lottery search for integer and transfer nft for wallet address), if we move this pattern to uikit we need to refactor all of the usages and write rules for each of them.

Reply:
Yes, good catch.

Reply:
Wait, you added it to the BalanceInput which is fine.

Reply:
I think it is ready to merge wdyt?